libyul/optimiser/ExpressionSplitter.cpp

```diff
@@ -46,14 +46,14 @@ void ExpressionSplitter::run(OptimiserStepContext& _context, Block& _ast)
 
 void ExpressionSplitter::operator()(FunctionCall& _funCall)
 {
-	vector<bool> const* literalArgs = nullptr;
+	vector<std::optional<LiteralKind>> const* literalArgs = nullptr;
 
 	if (BuiltinFunction const* builtin = m_dialect.builtin(_funCall.functionName.name))
 		if (builtin->literalArguments)
 			literalArgs = &builtin->literalArguments.value();
 
 	for (size_t i = _funCall.arguments.size(); i > 0; i--)
-		if (!literalArgs || !(*literalArgs)[i - 1])
+		if (!literalArgs || !(*literalArgs)[i - 1].has_value())
```

Comment:
I know `operator!` is an alternative to `!has_value()`, but it seems more explicit to me. Can remove it too.